File: engine/Kogase.Engine/Models/ProjectUser.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Kogase.Engine.Models
{
    public class ProjectUser
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public int ProjectId { get; set; }
        
        [ForeignKey("ProjectId")]
        public Project? Project { get; set; }
        
        [Required]
        public int UserId { get; set; }
        
        [ForeignKey("UserId")]
        public User? User { get; set; }
        
        [Required]
        public string Role { get; set; } = "Contributor"; // "Admin" or "Contributor"
    }
} 

File: backend/models/project_user.go
package models

import (
	"github.com/google/uuid"
)

// ProjectUser represents the join table between projects and users
type ProjectUser struct {
	ProjectID uuid.UUID `json:"project_id" gorm:"type:uuid;primaryKey"`
	UserID    uuid.UUID `json:"user_id" gorm:"type:uuid;primaryKey"`
	Role      string    `json:"role" gorm:"not null;default:contributor"` // contributor or admin
	Project   Project   `json:"-" gorm:"foreignKey:ProjectID"`
	User      User      `json:"-" gorm:"foreignKey:UserID"`
}
